### DISCUSSION TITLE
How about splitting HW SUPPORT in wifibox-alpine(5) into separate sections?

### DISCUSSION THREAD

**Discussion by rymdbar** [pgj/freebsd-wifibox#102] [General]
I'm glad to finally report that wifibox enables networking through a Broadcom BCM4360. It took me a few attempts though. Including installing alpine on bare-metal and reinstalling FreeBSD again, just to verify the interface actually works with linux.

Thanks for this enabling software, and good lateral thinking for coming up with the idea behind it!

For me, who don't come up with ideas on how to leverage a hypervisor to pragmatically solve problems it wasn't designed for, it was far from obvious that one needs to build the port from source in order to get a wl kernel module popping into existance.

A friendly suggestion is to split the HARDWARE SUPPORT section in wifibox-alpine(5) into a section of interfaces known to be working out-of-the box with binary pkg installation, and a section that in some way directs to the already good but sligthly hard-to-find instructions at: https://github.com/pgj/freebsd-wifibox-port.

Specifically, my environment is an Apple Macbook Pro A1502 EMC 2678 (MacBookPro11,1) with a 14e4:43a0, running FreeBSD 14.0-RELEASE on amd64 and ports from git 35dd96290a67 (wifibox-1.4.2, wifibox-alpine-20240506, wifibox-core-0.13.0).

**Comment by pgj** [comment]
It is my pleasure to help fellow FreeBSD users with bridging the gap around the wireless support.  Although I have to confess that the idea is not entirely mine.  I only contributed the "product" wrapper, that is making the installation easier and accessible for a larger audience.  But as your comment shows there is still room for improvements!

I admit that the `wl` driver support is not so easily accessible.  The main reason is due to licensing.  In my humble understanding, Broadcom would not let either me or the FreeBSD Project distribute the binary versions of driver directly.  That is why I have put it behind a port configuration option and decided not to dedicate a flavor for that for the binary packages.

I believe https://github.com/pgj/freebsd-wifibox-port/blob/main/README.md would not be needed for most of the cases.  It exists mostly because I like to document my work :-P  Jokes aside, the things are mentioned in the description will also be enforced by the port itself.  For example, the presence of the `linux64` kernel module and `/compat/linux` directory is checked in the `Makefile` and user will be notified if any of them is missing.  The same goes for the rest.  That said, you should not be excluded from any special knowledge if you simply tried to run `make` in the port's directory.

However, I agree that it would be nice if the `wifibox-alpine(5)` manual page gave a hint for the users about some of the drivers not being part of the default package.  It can be indeed confusing.  I will craft a PR for that and get back to you about it shortly.

**Reply by pgj** [reply]
https://github.com/pgj/freebsd-wifibox-alpine/pull/22

**Comment by pgj** [comment]
The manual page has been modified to contain the missing information.  This will be made part of the next release.
